appinventor/components/src/com/google/appinventor/components/runtime/Ev3GyroSensor.java

```diff
@@ -172,18 +192,13 @@ private double getSensorValue(String functionName) {
                        0,
                        sensorPortNumber,
                        SENSOR_TYPE,
-                       mode);
+                       mode.toInt());
   }
 
-  private void setMode(String newModeString) {
-    if (SENSOR_MODE_ANGLE_STRING.equals(newModeString))
-      mode = SENSOR_MODE_ANGLE;
-    else if (SENSOR_MODE_RATE_STRING.equals(newModeString))
-      mode = SENSOR_MODE_RATE;
-    else
-      throw new IllegalArgumentException();
-
-    this.modeString = newModeString;
+  private void setMode(GyroSensorMode newMode) {
+    // TODO: I imagine this function should reset the previousValue, similar to how the ColorSensor
```

Comment:
I believe that you're correct. The checker logic uses -1 to indicate that data has yet to be received. Given that changing the mode changes the value space of the data being received it really ought to reset the previous value to indicate no data has been received in the new value space. Otherwise, the comparisons occurring in that code are really apples to oranges (or angles to rates). I would recommend not changing it here and filing a bug instead on the main repo.

Reply:
Done added issue [here](https://github.com/mit-cml/appinventor-sources/issues/2291). I also added the issue number in the source.